File: Student.java
import java.util.Scanner;

public class Student {

	// array to store the grade of 5 student

	public final static int STUD_NUM = 5;
	public static int grades[] = new int[STUD_NUM];

	// method to get the sum of 5 student grades
	public static int getTotal(){

		int sum = 0;

		for(int i : grades)
			sum += i;

		return sum;
	}

	// method to get the average of 5 students 
	public static double getAverage(){

		int total = getTotal();

		double avg = (double) total / STUD_NUM;

		return avg;
	}

// main driver method 
	public static void main(String[] args) {
		
		// to read user input 
		Scanner input = new Scanner(System.in);

		// loop to read grades of 5 students 
		for(int i = 0; i < STUD_NUM; i++){

			System.out.print("Enter your grades: ");
			grades[i] = input.nextInt();

		}

		System.out.println("Total : "+ getTotal());
		System.out.println("Average : "+ getAverage());

	}

}

File: student.py


class Student:

	def __init__(self):
		self.name = ""
		self.marks = []
		self.avg = 0.0

	def calcAvg(self):

		sum = 0
			# sum up all the grades
		for i in self.marks:
			sum += i

		# calculate the average 
		self.avg = sum / len(self.marks)

		return self.avg

	# method to return the grade letter 
	def getGrade(self):
		self.calcAvg()
	# change the grades and ranges as per your own convenience
		if self.avg >= 90:
			return "A"
		elif self.avg >= 80:
			return "B"
		elif self.avg >= 70:
			return "C"
		elif self.avg >= 60:
			return "D"
		elif self.avg < 60:
			return "F"

	def readMarksAndName(self):
		self.name = input("\nEnter student name: ")
		for i in range(5):
			while True:
				mark = float(input("Enter ex {ind} marks ".format(ind = (i + 1))))
				if(mark > 0 and mark < 110):
					self.marks.append(mark)
					break
				else:
					print("Invalid marks!!")


	def sortAsc(self):
		marks.sort()

marks = []
students = []

def display():

	print("\nName \t Marks\t\t\t\tAverage\t\tGrade")
	print("\n**********************************************")
	for stud in students:	
		print(stud.name,"\t", stud.marks, "\t", stud.calcAvg(), "\t\t",stud.getGrade())

def main():
	
	numStudent = int(input("Enter number of students: "))

	# list for storing student names 

	for i in range(numStudent):
		std = None
		std = Student()
		std.readMarksAndName()
		students.append(std)

	display()

main()


